assets/javascripts/discourse/controllers/admin-plugins-explorer.js.es6

```diff
@@ -117,16 +127,41 @@ export default Ember.Controller.extend({
       .finally(() => this.set("loading", false));
   },
 
+  _readQueryFile(file) {
+    const reader = new FileReader();
+    reader.onload = (evt) => {
+      this._importQuery(evt.target.result);
+    };
+
+    reader.readAsText(file);
+  },
+
+  _importQuery(file) {
+    const query = JSON.parse(file).query;
```

Comment:
I think we should add some JSON parse error handling here, in case the JSON in the file is not correctly structured. Then you can show a message to the user and reset the `loading` state

Reply:
Also do we know what the structure should look like ahead of time? Maybe we should check for the keys that are required and throw an error if the user doesn't provide them?